akka-stream/src/main/scala/akka/stream/impl/fusing/IteratorInterpreter.scala

```diff
@@ -145,7 +145,8 @@ private[akka] class IteratorInterpreter[I, O](val input: Iterator[I], val ops: S
       outHandlers,
       logics,
       (_, _, _) ⇒ throw new UnsupportedOperationException("IteratorInterpreter does not support asynchronous events."),
-      fuzzingMode = false)
+      fuzzingMode = false,
+      null)
```

Comment:
@drewhk Thanks for the explanation, Endre! (I appreciate it!)


I think it boils down to two things: hard-to-maintain code (i.e. mutable state etc) should be created when performance matters and the harder-to-maintain solution has been benchmarked to be significantly (contextual) better than an easier-to-maintain solution, and in which case there should be both a bench AND comments in the code explaining why the code is there, at the very least to console the maintainer that they're having a rougher time for a reason.

and

`null` should be used when defensible, which ideally also should be documented. :)